### PR TITLE
Expectation ftp v21

### PR DESCRIPTION
**Pull request by regit**
Update of #3003 

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/550

Describe changes:
- Address comments
- Add memcap to ftp

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/353
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/137

### CODE REVIEW

File: src/app-layer-ftp.c [hunk 1]
```diff
@@ -54,11 +54,136 @@
...
+
+    if (unlikely(ptr == NULL))
+        return NULL;
+
+    FTPIncrMemuse((uint64_t)(n * size));
+
+    return ptr;
+}
+
+static void *FTPRealloc(void *ptr, size_t orig_size, size_t size)
```
Comment (victorjulien) [inline src/app-layer-ftp.c]: looking at this it seems shrinking actually won't work correctly wrt memcaps. Same for the original HTPRealloc version.

Comment (regit) [reply src/app-layer-ftp.c]: Indeed. Want me to fix it or do you take it ?

Comment (victorjulien) [reply src/app-layer-ftp.c]: Looks like ReassembleRealloc is a better example to copy. Can you take it for FTP?

Comment (regit) [reply src/app-layer-ftp.c]: Yes, sure.

**Pull request closed without merging**